From fa51b68eeb27fd25c7a4a25b627215d62ccb058e Mon Sep 17 00:00:00 2001
From: EdenEast <edenofest@gmail.com>
Date: Tue, 13 Feb 2024 13:41:52 -0500
Subject: [PATCH] 10 degree angle for stand

---
 src/lagrange_keyboard/core.clj | 36 +++++++---------------------------
 1 file changed, 7 insertions(+), 29 deletions(-)

diff --git a/src/lagrange_keyboard/core.clj b/src/lagrange_keyboard/core.clj
index ee91444..765f051 100644
--- a/src/lagrange_keyboard/core.clj
+++ b/src/lagrange_keyboard/core.clj
@@ -235,12 +235,12 @@
 ;; real estate, although this should be avoided, unless a bridge is
 ;; installed.
 
-(def stand-tenting-angle (degrees 35))   ; The angle of extrusion.
+(def stand-tenting-angle (degrees 10))   ; The angle of extrusion.
 (def stand-shape-factor 0)               ; 0 is radial extrusion, 1 is projection.
 (def stand-width 25/2)                   ; The width of the strip forming the stand cross-section.
 (def stand-minimum-thickness [6 4 4/10]) ; Thickness at inner bottom, inner top and outer bottom.
-(def stand-boss-indexes [0 3, 1 7])      ; The screw bosses shared by the stand.
-(def stand-split-points [15 51])         ; The locations where the arms begin.
+(def stand-boss-indexes [0 3, 2 8, 1 7]) ; The screw bosses shared by the stand.
+(def stand-split-points [7 60])          ; The locations where the arms begin.
 (def stand-arm-lengths [3 11, 2 6])      ; Hom much to shorten the arms.
 (def stand-arm-slope (degrees 13))
 
@@ -1781,7 +1781,7 @@
     (boss-place x d endpoints (translate [0 0 -3/2]
                                          (countersink (/ cover-countersink-diameter 2)
                                                       cover-countersink-height
-                                                      50 20)))))
+                                                      50 50)))))
 
 ;;;;;;;;;;;;
 ;; Bridge ;;
@@ -2374,31 +2374,9 @@
 
               (let [parts (list*
                            (case-placed-shapes stand-boss)
-                           (first hulled-parts)
-
-                           (for [i [1 3 0 2]
-                                 :let [q (quot i 2)
-                                       r (rem i 2)
-                                       qq (- 1 q q)
-                                       rr (- 1 r r)
-
-                                       [take-this take-that] (cond->> [take take-last]
-                                                               (pos? q) reverse)]]
-                             (difference
-                              (apply union (nth hulled-parts (inc i)))
-
-                              (->> (sphere 1/2)
-                                   (translate [0 0 (* rr (+ 1/2 (stand-minimum-thickness r)))])
-                                   (stand-section (- 1 r)
-                                                  (for [n [-100 100]
-                                                        [t z] [[(/ wall-thickness -2) 0]
-                                                               [100 (* (Math/tan stand-arm-slope) 100)]]
-                                                        z_add [0 100]] [n (* qq t) (* rr (+ z z_add))]))
-                                   (take-this (inc (stand-arm-lengths i)))
-                                   (take-that 1)
-                                   (apply hull)))))
-
-                    [upper-parts lower-parts] (split-at 4 parts)]
+                           (first hulled-parts))
+
+                    [upper-parts lower-parts] (split-at 1000 parts)]
 
                 (apply union
                        (difference
-- 
2.35.0.windows.1

